design-docs/improved-console.md

```diff
@@ -0,0 +1,422 @@
+# Improved Console Output
+
+## Primary Goals
+* Improve perceived performance by making user aware of background work
+* Modern feel
+* User extensibility
+
+### User Goals for Console
+The user wants to answer the following questions:
+
+* When will my build finish?
+    * Soon (should I keep watching it) or later ("I'll check back")
```

Comment:
Gradle doesn't yet know when it'll finish, but it does know if it's working on the last task. That is something we're not surfacing to the user and should.